### PR TITLE
Feat/estimate fee correct version

### PR DESCRIPTION
**Pull request by tdelabro**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
- Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

## What is the current behavior?

Resolves: #NA

## What is the new behavior?

-
-
-

### CODE REVIEW

File: crates/client/rpc/src/errors.rs [hunk 1]
```diff
@@ -19,28 +19,30 @@ pub enum StarknetRpcApiError {
...
     #[error("Invalid contract class")]
     InvalidContractClass = 50,
-    #[error("Too many storage keys requested")]
-    ProofLimitExceeded = 10000,
-    #[error("Too many keys provided in a filter")]
-    TooManyKeysInFilter = 34,
+    #[error("Class already declared")]
+    ClassAlreadyDeclared = 51,
+    #[error("The transaction version is not supported")]
+    UnsupportedTxVersion = 61,
```
Comment (tdelabro) [inline crates/client/rpc/src/errors.rs:39]: Did you choose those numbers yourself?

File: crates/client/rpc/src/lib.rs [hunk 2]
```diff
@@ -528,9 +528,17 @@ where
         request: Vec<BroadcastedTransaction>,
         block_id: BlockId,
     ) -> RpcResult<Vec<FeeEstimate>> {
-        // TODO:
-        //      - modify BroadcastedTransaction to assert versions == "0x100000000000000000000000000000001"
-        //      - to ensure broadcasted query signatures aren't valid on mainnet
+        let smthng = request.iter().any(|tx| match tx {
```
Comment (tdelabro) [inline crates/client/rpc/src/lib.rs:531]: `smthng` is not a good name

File: crates/client/rpc/src/lib.rs [hunk 3]
```diff
@@ -528,9 +528,17 @@ where
...
+            BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction::V0(tx_v0)) => tx_v0.is_query,
+            BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction::V1(tx_v1)) => tx_v1.is_query,
+            BroadcastedTransaction::Declare(BroadcastedDeclareTransaction::V1(tx_v1)) => tx_v1.is_query,
+            BroadcastedTransaction::Declare(BroadcastedDeclareTransaction::V2(tx_v2)) => tx_v2.is_query,
+            BroadcastedTransaction::DeployAccount(deploy_tx) => deploy_tx.is_query,
+        });
+
+        if smthng {
+            return Err(StarknetRpcApiError::UnsupportedTxVersion.into());
+        }
```
Comment (tdelabro) [inline crates/client/rpc/src/lib.rs:541]: Why are those not supported for `estimate_fee`?

File: crates/client/rpc/src/errors.rs (hunk 1 above)
Comment (gaetbout) [reply crates/client/rpc/src/errors.rs:39]: Comes from:
https://github.com/starkware-libs/starknet-specs/blob/0e859ff905795f789f1dfd6f7340cdaf5015acc8/api/starknet_write_api.json#L272

File: crates/client/rpc/src/lib.rs (hunk 3 above)
Comment (gaetbout) [reply crates/client/rpc/src/lib.rs:541]: What do you mean, isn't the point of the issue to block/prevent all the one that aren't query_version?

Comment (gaetbout) [reply crates/client/rpc/src/lib.rs]: These are the cases:
 - Deploying: `__validate_deploy__` should ensure tx version is  2**128 + 1
 - Invoke: `__validate_` should ensure tx version is  2**128 + 1
 - Declaring: `__validate_declare__` should ensure tx version is  2**128 + 2
 
I think both BroadcastedInvokeTransaction::V0 and BroadcastedDeclareTransaction::V1 shouldn't even be supported 
so the another error should be thrown for those?

Comment (gaetbout) [reply crates/client/rpc/src/lib.rs]: ref:
https://docs.starknet.io/documentation/starknet_versions/version_notes/#:~:text=New%20transaction%20version%20(version%201)%20for%20invoke%20and%20declare%20transactions%3A

Comment (gaetbout) [reply crates/client/rpc/src/lib.rs]: https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/Blocks/transactions/

Comment (gaetbout) [reply crates/client/rpc/src/lib.rs]: Still working on this, so just ignore it at this point :)

Comment (gaetbout) [reply crates/client/rpc/src/lib.rs]: GM @tdelabro , @apoorvsadana ,
As stated in the documentation 
https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/Blocks/transactions/
Invoke v0 and declared v1 are deprecated and will be removed.
Should those already return some different error?
Or should this be part of another issue for when they will effectively be removed?

File: crates/client/rpc/src/errors.rs (hunk 1 above)
Comment (tdelabro) [reply crates/client/rpc/src/errors.rs:39]: maybe mention it in a comment

**Pull request closed without merging**